### PR TITLE
[Merged by Bors] - cmd/node: use shared filelock by default

### PR DESCRIPTION
**Pull request by dshulyak**
originally i added file lock for a specific data directory, as it seemed nicer that we can run multiple instances of go-spacemesh. however there are ongoing reports that people somehow run more than one, and submit different atxs, etc.

it would be very stupid if we will have malfeasence proofs because of that. so i would rather request people to configure separate file locks if they wish to run many instances on same computer, to do so there is an option `--filelock`, that defaults to `/tmp/spacemesh.lock`

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -154,6 +155,7 @@ func DefaultTestConfig() Config {
 func defaultBaseConfig() BaseConfig {
 	return BaseConfig{
 		DataDirParent:       defaultDataDir,
+		FileLock:            "/tmp/spacemesh.lock",
```
Comment (pigmej) [inline config/config.go]: I think we need to make it platform independent?

**Review by pigmej — CHANGES_REQUESTED** [review]

File: config/config.go (hunk 1 above)
Comment (dshulyak) [reply config/config.go]: yeah i realized that it is probablyy no ok on windows. will look up what works there

Comment (dshulyak) [reply config/config.go]: actually there was os.TempDir

**Review by pigmej — APPROVED** [review]

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors merge

**Comment by pigmej** [comment]
bors try

**Pull request closed without merging**